sw/device/lib/drivers/dma/dma.c

```diff
@@ -675,8 +675,17 @@ dma_config_flags_t dma_load_transaction( dma_trans_t *p_trans )
     ////printf("src ptr - Wrote %08x @ ----\n", dma_cb.trans->src->ptr );
     //printf("dst ptr - Wrote %08x @ ----\n", dma_cb.trans->dst->ptr );
     dma_cb.peri->SRC_PTR = dma_cb.trans->src->ptr;
-    dma_cb.peri->DST_PTR = dma_cb.trans->dst->ptr;
-    
+
+    if(dma_cb.trans->mode != DMA_TRANS_MODE_ADDRESS) {
+        /*
+            Write to the destination pointers only if we are not in address mode,
+            otherwise the destination address is read in a separate port in parallel with the data
+            from the address port
+        */
+        dma_cb.peri->DST_PTR = dma_cb.trans->dst->ptr;
```

Comment:
The agreed format in the FW team was to use the curly braces in their own line. We should correct these occurrences for consistency (at least in the HAL, the examples are not so consistent on this, they should go on a refactor at some point)

Reply:
how shall I modify them?

Reply:
We should do: 
```C
    if(dma_cb.trans->mode != DMA_TRANS_MODE_ADDRESS) 
    {
        /*
            Write to the destination pointers only if we are not in address mode,
            otherwise the destination address is read in a separate port in parallel with the data
            from the address port
        */
        dma_cb.peri->DST_PTR = dma_cb.trans->dst->ptr;
     } 
     else 
     {
        dma_cb.peri->ADDR_PTR = dma_cb.trans->src_addr->ptr;
     }
```